analyzers/src/SonarAnalyzer.CSharp/Rules/ExceptionsShouldBeLogged.cs

```diff
@@ -0,0 +1,119 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class ExceptionsShouldBeLogged : SonarDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S6667";
+    private const string MessageFormat = "Logging in a catch clause should include the exception.";
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(c =>
+            {
+                var catchClauseSyntax = (CatchClauseSyntax)c.Node;
+                var walker = new CatchLoggingInvocationWalker(c.SemanticModel);
+                if (walker.SafeVisit(catchClauseSyntax) && !walker.IsExceptionLogged)
+                {
+                    foreach (var invocation in walker.LoggingInvocations)
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, invocation.GetLocation()));
```

Comment:
I don't like that it raises multiple issues in a single `catch` block. Though I don't have a good idea to change this. Is it possible to add multiple secondary locations to an issue?